### PR TITLE
Add simple Journal model and create upon rota generation

### PR DESCRIPTION
**Pull request by chriscarter90**
* `RotaGenerationLogEntry` model shows a history of rota generation runs
* Yet to be put on the UI (next PR)

### CODE REVIEW

**Comment by chriscarter90** [comment]
@oskarpearson: Seeing as we spoke about this the other day, would be good to get your input on this too?

File: app/models/journal_entry.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class JournalEntry < ActiveRecord::Base
```
Comment (ansonK) [inline app/models/journal_entry.rb]: Would a better name for this be "RotaGenerationLog" as it is tightly bound to that.
JournalEntry is pretty generic .. ?

File: app/jobs/generate_new_rota.rb [hunk 2]
```diff
@@ -1,15 +1,30 @@
+require_relative "../../lib/rota_generation"
+
 class GenerateNewRota < Que::Job
```
Comment (ansonK) [inline app/jobs/generate_new_rota.rb]: This should extend ActiveJob::Base so we can switch out the actual job runner without needing to change the job implementation

File: app/jobs/generate_new_rota.rb [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+    )
+
+    RotaGeneration::Generator.new(
+      empty_rota_slots,
       member_ids
     ).generate_rota
 
+    success = false
+
     RotaSlot.transaction do
```
Comment (ansonK) [inline app/jobs/generate_new_rota.rb]: We can use ActiveRecord::Base.transaction here instead of RotaSlot

File: db/structure.sql [hunk 4]
```diff
@@ -30,6 +30,39 @@ SET default_tablespace = '';
...
+    end_time timestamp without time zone,
+    status character varying
+);
+
+
+--
+-- Name: journal_entries_id_seq; Type: SEQUENCE; Schema: public; Owner: -
+--
+
+CREATE SEQUENCE journal_entries_id_seq
```
Comment (oskarpearson) [inline db/structure.sql]: Do we need a sequence here? We could just use the primary key, right? Might be simpler.

**Comment by oskarpearson** [comment]
Overall I like it!

I think it'd be simplified if you could just use the ID field and get rid of the sequence number. Sequence numbers are Postgres specific, so it limits our ability to run on other systems.

It's also occurred to me that it might be useful to log the user that actually created the rota item. This way if someone creates a rota entry that's very time consuming, you can track down who did it and speak to them.

**Comment by chriscarter90** [comment]
@oskarpearson I've added the user-entry relationship now, thanks! I'm not sure what you mean about the sequence number though?

@ansonK We should talk tomorrow about your last comment? I'm still unsure of how we can make that change if the method is named differently in `ActiveJob`?

File: db/structure.sql (hunk 4 above)
Comment (chriscarter90) [inline db/structure.sql]: Ah this is the sequence in question. Not sure how to get rid of it? Never used this `structure.sql` milarky.

**Comment by chriscarter90** [comment]
screen shot 2015-07-07 at 14 53 51

**Pull request merged**